### PR TITLE
Add license-checker on CI

### PR DESCRIPTION
**Pull request by allevo**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

Fix https://github.com/fastify/fastify/issues/1356

The output si available here: https://travis-ci.org/fastify/fastify/jobs/472663107#L569
There's a summary also (see below).

Are there some cases on which the build fails? For example with Apache Licence?
The license is checker when the dependency is added but if the author doesn't make a breaking change on changing license, I'd like to have a check that fails.

For time being fastify has:
```
├─ MIT: 38
├─ ISC: 4
├─ BSD-3-Clause: 2
└─ BSD-2-Clause: 1
```

### CODE REVIEW

**Comment by mcollina** [comment]
Overall I would just whitelist those 4 licenses, and fail with anything else. Then, we add more if it’s the case.

**Comment by allevo** [comment]
Failing test:
https://travis-ci.org/fastify/fastify/jobs/472726132#L572

**Comment by mcollina** [comment]
There is something wrong in your script, as those licenses should be matched. Anyway, you can use the —onlyAllow option of license-checker, it already implements this behavior.

**Comment by allevo** [comment]
done

**Comment by allevo** [comment]
Done

**Comment by allevo** [comment]
Added

**Comment by delvedor** [comment]
We should also update the README :)

Question: add o remove a license, implies a semver major?

**Comment by allevo** [comment]
IMHO We shouldn't. `fastify` doesn't incorporate dependency licenses. But I'm not a lawyer 😄

**Comment by delvedor** [comment]
I think that the point to use this tool is to guarantee that we use only a restricted list of licenses.
We should document it *(before merging the pr)* and have a discussion *(in the future is ok as well)* regarding the versioning.

**Comment by allevo** [comment]
Added to README

File: README.md [hunk 1]
```diff
@@ -52,7 +52,7 @@ fastify.get('/', (request, reply) => {
 })
 
 // Run the server!
-fastify.listen(3000, (err, address) => {
+fastify.(3000, (err, address) => {
```
Comment (delvedor) [inline README.md]: I guess this is a typo ;)
```suggestion
fastify.listen(3000, (err, address) => {
```

File: README.md [hunk 2]
```diff
@@ -208,3 +208,10 @@ Past Sponsors:
 ## License
 
 Licensed under [MIT](./LICENSE).
+
+The `fastify` CI guarantees the accepted dependencies licenses are:
```
Comment (delvedor) [inline README.md]:
```suggestion
The `fastify` CI guarantees that the accepted dependencies licenses are:
```

File: README.md (hunk 1 above)
Comment (allevo) [reply README.md]: ah, nice... sorry!

File: README.md [hunk 3]
```diff
@@ -208,3 +208,10 @@ Past Sponsors:
 ## License
 
 Licensed under [MIT](./LICENSE).
+
+The `fastify` CI guarantees that the accepted dependencies licenses are:
+- MIT
+- ISC
+- BSD-3-Clause
+- BSD-2-Clause
+- Apache-2.0
```
Comment (mcollina) [inline README.md]: I would include some different text here. We do not imply that we won’t accept dependencies under a different license, but we’ll just vet them.

Note, we have no dependencies on Apache-2, so it should be removed. This is a whitelist of what we have right now, not what we would accept in the future. As such, it should be kept to the minimum.

File: package.json [hunk 4]
```diff
@@ -14,7 +14,8 @@
     "coverage": "npm run unit -- --cov --coverage-report=html",
     "coveralls": "npm run unit -- --cov",
     "bench": "branchcmp -r 2 -g -s \"npm run benchmark\"",
-    "benchmark": "npx concurrently -k -s first \"node ./examples/simple.js\" \"npx autocannon -c 100 -d 5 -p 10 localhost:3000/\""
+    "benchmark": "npx concurrently -k -s first \"node ./examples/simple.js\" \"npx autocannon -c 100 -d 5 -p 10 localhost:3000/\"",
+    "license-checker": "license-checker --production --onlyAllow='MIT;ISC;BSD-3-Clause;BSD-2-Clause;Apache-2.0'"
```
Comment (mcollina) [inline package.json]: If we have no dependency on Apache-2, let’s remove it. It’s a whitelist because there are too many licensenses out there.

**Comment by allevo** [comment]
I've no idea how to write that down. If someone has a better idea, I'll address the suggestion.

File: README.md [hunk 5]
```diff
@@ -208,3 +208,9 @@ Past Sponsors:
 ## License
 
 Licensed under [MIT](./LICENSE).
+
+`fastify` team cares about licence. We list all dependencies licences in order to stay as "open" as possibile:
```
Comment (mcollina) [inline README.md]:
```suggestion
For your convenience, here is a list of all the licenses of our production dependencies:
```

**Pull request merged**